### ISSUE TITLE
Attending to full mel or prenet/dropout mel? 

### ISSUE THREAD

**Issue by yxt132** [NVIDIA/tacotron2#131]
I saw Rafael made change in "model.py" in May 2018 so that the model attends to the full mel rather than the prenet and dropout mel. However, the code was modified in November 2018 and it now attends to the prenet again. What is the reason for changing it back? Rafael previously indicated that changing the attention to full mel can speed up the convergence. What made you change the idea? Thanks.

**Comment by rafaelvalle** [comment]
The model generalizes better if it attends to the prenet Mel.

**Comment by yxt132** [comment]
Thank you for your reply, Rafael. I am currently training on the World English Bible corpus with about 25,000 utterances in the trainging set. The sampling rate is 12,000. I used the latest repo to train this dataset with all the default hyper parameters except the sampling rate. I have only one nvidia Geforce 2080 ti with 12 G memory so that I have to cut the batch size to 4 for my case. I was not able to get meaningful alignment even after 250,000 iterations. Then I decreased the learning rate from 1e-3 to 1e-4. The validation loss decreased sharply (don't know why). The loss is still decreasing but the alignment is still bad. What could be the reason in your opinion. Should I change the mel-max from 8000 to 6000 (half the sr) and decrease the window length/hop length? What's your recommendation? I could do trial and error that all of that takes  significant time due to the limitation of my computing resources. I have run the program for 4 days now to get to 300,000 iterations. Below is the alignment and loss figures. Any suggesions are greatly appreciated!
![image](https://user-images.githubusercontent.com/43161328/51420701-651b9f00-1bcf-11e9-9475-c9d5f614c455.png)
![image](https://user-images.githubusercontent.com/43161328/51420709-75cc1500-1bcf-11e9-9334-35153644c296.png)

**Comment by rafaelvalle** [comment]
What is the distribution of sentence lenghts in seconds from your dataset?
What GPU are you using?

**Comment by yxt132** [comment]
The utterance lengths range from 3-12 seconds roughly. I used Nvidia Geforce 2080 TI. I tried to cut the leading and trailing sliences and it helped. I started to get some meaningful alignment at 180,000 iterations but only after 300,000 iterations the alignments are acceptable. I was wondering if it is because of the audio quality (i.e., low sampling rate).

**Comment by rafaelvalle** [comment]
It can be due to the dataset quality. 
You can also train starting with the weights from the pre-trained model, keeping the embedding layer from the model that will be trained. It should learn attention faster.

**Comment by rafaelvalle** [comment]
Closing due to inactivity.
